specification/sections/01-flag-evaluation.md

```diff
@@ -171,6 +171,16 @@ client.getMetadata().getDomain(); // "domain-1"
 In previous drafts, this property was called `name`.
 For backwards compatibility, implementations should consider `name` an alias to `domain`.
 
+#### Requirement 1.2.3
+
+> The client interface **MUST** define a `all flag metadata` member or accessor, returning a collection of flag metadata from the provider.
+
+```typescript
+client.getAllFlagMetadata() // [{"key": "featureA", "type": "boolean"}, {"key": "featureB", "type": "string"}]
```

Comment:
I wonder if this should be an object that contains a list of flags.

```ts
{
  flags: [{"key": "featureA", "type": "boolean"}, {"key": "featureB", "type": "string"}]
}
```

That would allow us to easily add new metadata to the response in a non-breaking way. For example, a last updated timestamp and version property would be extremely useful.

Reply:
Makes sense to me, I'll make that update and see what others think.